extensions/amp-image-slider/0.0/amp-image-slider.js

```diff
@@ -0,0 +1,566 @@
+   */
+  dragMove(e) {
+    e.preventDefault();
+    e.stopPropagation(); // avoid clashing with clickImage
+    this.pointerMoveX_(e.pageX);
+  }
+  /**
+   * Remove listeners on drag end
+   * e is optional since dragEnd will also be used for unregister cleanup
+   * @param {Event|null} e
+   */
+  dragEnd(e) {
+    if (e) {
+      e.stopPropagation();
+    }
+
+    this.win.removeEventListener('mousemove', this.dragMove);
+    this.win.removeEventListener('mouseup', this.dragEnd);
+
+    this.moveOffset_ = 0;
+    this.splitOffset_ = 0;
+  }
+  /**
+   * Add listeners on drag start
+   * @param {Event} e
+   */
+  touchStart(e) {
+    this.container_.addEventListener('touchmove', this.touchMove);
+    this.container_.addEventListener('touchend', this.touchEnd);
+
+    this.moveOffset_ = e.touches[0].pageX;
+    this.splitOffset_ = this.bar_./*OK*/getBoundingClientRect().left;
+  }
+  /**
+   * Handle touch move
+   * @param {Event} e
+   */
+  touchMove(e) {
+    // When we are holding bar button (or its icon)
+    // Do not scroll page
+    if (e.target === this.barButton_ ||
+        (this.barButtonIcon_ && e.target === this.barButtonIcon_)) {
+      e.preventDefault();
+    }
+    e.stopPropagation(); // avoid clashing with clickImage
+    if (e.touches.length > 0) {
+      this.pointerMoveX_(e.touches[0].pageX);
+    }
+  }
+  /**
+   * Cleanup when touch released
+   * @param {Event|null} e
+   */
+  touchEnd(e) {
+    if (e) {
+      // Avoid bubbling up to element
+      e.stopPropagation();
+    }
+    this.container_.removeEventListener('touchmove', this.touchMove);
+    this.container_.removeEventListener('touchend', this.touchEnd);
+
+    this.moveOffset_ = 0;
+    this.splitOffset_ = 0;
+  }
+
+  /**
+   * Pointer move X logic
+   * @param {number} pointerX
+   * @private
+   */
+  pointerMoveX_(pointerX) {
+    // const {width} = this.getLayoutBox();
+    const {width} = this.container_./*OK*/getBoundingClientRect();
+    const {left: leftBound, right: rightBound}
+    //    = this.getLayoutBox();
+        = this.container_./*OK*/getBoundingClientRect();
+
+    const moveX = pointerX - this.moveOffset_;
+    const newPos = Math.max(leftBound,
+        Math.min(this.splitOffset_ + moveX, rightBound));
+    const newPercentage = (newPos - leftBound) / width;
+    this.updatePositions(newPercentage);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    // https://github.com/ampproject/amphtml/pull/16688
+    user().assert(isExperimentOn(this.win, 'amp-image-slider'),
+        'Experiment <amp-image-slider> disabled');
+
+    // amp-img initially have no size as amp-image-slider
+    // is not yet completely built.
+    // scheduleLayout without setAsOwner seems work
+    // as a hint that the new layout should be inspected
+
+    // Actually, now they are no longer useful
+    // But kept since there is an layer experiment check
```

Comment:
I'm a bit confused by what your comment means. Is this necessary for layers when we don't call `setAsOwner`? Is this ever necessary?